[Woo POS] M2: Cart Animation

staskus opened the pull request:
<!-- Remember about a good descriptive title. -->

Part of: #13444
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

Adding Cart animation when switching between order building (product list), and order finalizing (checkout) states.

Cart animation is required for further transitions when switching between 'Ready for Payment' -> 'Payment Processing' states, so it's a good time to prepare a basic animation and view structure that supports transitions.

## Solution

I restructured the view for `SwiftUI` to pick up transitions between views:
- The same `CartView` is always part of the dashboard `HStack`
- `ProductList` conditionally appears on the leading side with a transition
- `TotalsView` conditionally appears on the trailing side with a transition

Triggering animation when `viewModel.orderStage` changes.

## Testing information

1. Open Woo -> Menu -> Point of Sale Mode
2. Add items to cart
3. Check out
4. Confirm CartView and TotalsView transitions with animation
5. Come back
6. Confirm CartView and ProductListView transitions with animation

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->

https://github.com/user-attachments/assets/1ddc594c-2521-484a-97ab-fea5e312741b

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.
- [x] This PR includes refactoring; smoke testing of the entire section is needed.

staskus committed:
Add Cart animation when switching between order stages

dangermattic commented:
<!--
  0 Errors
  2 Warnings: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/216">19.8</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

joshheald reviewed (approved):
Works well, nice one!

These transitions will make a big difference to how easy it is to understand what's going on.

One minor thing I noticed – when the reader is connected, and you go back and forward without changing the selected products, the totals/subtotal table slides down a little as it's transitioning in. I guess this is due to the payments view above being added in during the transition, and we may need a placeholder to prevent it happening.

staskus commented:
@joshheald, thanks for the review!

> One minor thing I noticed – when the reader is connected, and you go back and forward without changing the selected products, the totals/subtotal table slides down a little as it's transitioning in

Yes, sliding down wouldn't be visible if we didn't have animations. However, it's connected to `cardReaderView` appearing/disappearing. When we enter the view, we start to collect payment which triggers changes of `isShowingCardReaderStatus`, which moves the `totalsFieldsView` down. I'll think if there's anything minor that could be changed to improve this.

staskus committed:
Disable animations from TotalsView at the moment it's appearing

It allows to avoid some totalFieldsView transition issues when animating appearing of TotalsView
